scripts/analytics/file_summary.py

```diff
@@ -27,12 +28,12 @@ def get_events(self, date):
 
         # Convert to a datetime at midnight for queries and the timestamp
         timestamp_datetime = datetime(date.year, date.month, date.day).replace(tzinfo=timezone.utc)
-        query_datetime = timestamp_datetime + timedelta(days=1)
 
         file_qs = OsfStorageFile.objects
 
-        public_query = Q(node__is_public=True)
-        private_query = Q(node__is_public=False)
+        quickfiles_query = Q(target_object_id__in=QuickFilesNode.objects.values('id'))
+        public_query = (Q(target_object_id__in=Node.objects.filter(is_public=True).values('id')) | quickfiles_query)
+        private_query = Q(target_object_id__in=Node.objects.filter(is_public=False).values('id'))
```

Comment:
For these guys to work reliably, you also need to make sure the `target_content_type` is `Node` so just need to add:

```
from django.contrib.contenttypes.models import ContentType

content_type = ContentType.objects.get_for_model(Node)
quickfiles_query = Q(target_object_id__in=QuickFilesNode.objects.values('id'), target_content_type=content_type)
```

Without the content type filter as well it might return some other things that aren't nodes that have the same ID as a node in that query.

Which makes me think, actually, we will probably need to tweak this script after the divorce and files can be on things other than nodes... but that's for another day!

Reply:
Ah, really good to know!